src/Microsoft.AspNetCore.Dispatcher/DispatcherBase.cs

```diff
@@ -78,17 +82,17 @@ public virtual IList<EndpointSelector> Selectors
                     return;
                 }
 
-                var selectorContext = new EndpointSelectorContext(httpContext, Endpoints.ToList(), Selectors);
+                var selectorContext = new EndpointSelectorContext(httpContext, Endpoints, Selectors);
```

Comment:
Be wary of changes that remove a `ToList()` or similar. This was an intentional defensive copy